File: js/bubble_sort.js
"use strict";
/**
 * Sorting problem:
 *
 * Input: A sequence of n numbers {a1, a2, ... , an }
 *
 * Output: A permutation (reordering) {a1', a2', ... , an' } 
 **/
function bubbleSort(arr) {

    let i = 0,
        arrLastIndex = arr.length -1,
        j,
        temp;

    for (; i < arrLastIndex; i++) {
        
        j = arrLastIndex;

        for (; j > 0; j--) {
            
            if (arr[j] < arr[j-1]) {
            
                temp = arr.slice(j,j+1)[0];
                arr[j] = arr[j-1];
                arr[j-1] = temp; 
            
            }
        
        }

    }

    return arr;

}

console.log(bubbleSort(generateArr(1,100)));
console.log(bubbleSort(generateArr(5,100)));
console.log(bubbleSort(generateArr(5,100)));
console.log(bubbleSort(generateArr(100,-1000)));
console.log(bubbleSort(generateArr(1001,100)));
console.log(bubbleSort(generateArr(105,100000)));
console.log(bubbleSort(generateArr(10000,100000000)));
console.log(bubbleSort(generateArr(10000,100000000, true)));


function generateArr(length, range, withdecimals) {
    var arr = [];

    while(arr.length < length) {

        var randomnumber = Math.ceil(Math.random()*range) + ((withdecimals) ? Math.random() : 0);
        arr[arr.length] = randomnumber;

    }

    return arr;

} 

File: algorithms/python/bubble_sort.py
import random as random
#Sorting problem:
#Input: A sequence of n numbers {a1, a2, ... , an }
#Output: A permutation (reordering) {a1', a2', ... , an' } 

def bubbleSort(arr):
    temp = None
    for i in xrange(0,len(arr)):
        for j in xrange(len(arr)-1,0,-1):
            if arr[j] < arr[j-1]:
                temp = arr[j]
                arr[j] = arr[j-1]
                arr[j-1] = temp
    return arr

def generateArr(length, rg, withdecimals):

    arr = []
    randomNumber = None

    while len(arr) < length:
        if withdecimals:
            randomNumber = random.randrange(rg)
        elif rg > 0:
           randomNumber = random.randint(0,rg)
        else:
           randomNumber = random.randint(rg,0)
        arr.append(randomNumber)
    return arr

print(bubbleSort(generateArr(1, 100, False)));
print(bubbleSort(generateArr(5, 100, False)));
print(bubbleSort(generateArr(5, 100, False)));
print(bubbleSort(generateArr(100, -1000, False)));
print(bubbleSort(generateArr(1001, 100, False)));
print(bubbleSort(generateArr(105, 100000, False)));
